pkg/volume/fc/fc_util.go

```diff
@@ -254,7 +254,12 @@ func (util *fcUtil) AttachDisk(b fcDiskMounter) (string, error) {
 		return "", err
 	}
 
-	return devicePath, nil
+	exists, err := mount.PathExists(devicePath)
+	if exists && err == nil {
+		return devicePath, nil
+	} else {
+		return "", err
```

Comment:
It should return an actual error instead of nil when `exists` is false and `err` is nil.

Reply:
@jsafrane do you want to say when `exists` is true and `err` is nil, then we should return err ?

Reply:
No, when `exists` is false (= returned device does not exist), then return error even when `err` is nil.

Reply:
@jsafrane Sorry for asking again, Do I have to make another condition ?
```
if exists == false && err == nil{
     return "", nil
}
```

Reply:
@jsafrane Ok, so I have to add 
```
if exists == false && err == nil{
    return "", err
}
```
If it is correct, why we are returning err at place of nil ? And why are we not doing it with `exists && err == nil`.

Reply:
No, if `exists == false`, you should return a *new error*, like `fmt.Errorf("device %s does not exist", devicePath)`. Why else do you add a check for `devicePath`, if you don't use the result?